File: typescript-client-sdk/src/sdk/models/operations/getcustomerscustomeridcredits.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform, Type } from "class-transformer";


export class GetCustomersCustomerIdCreditsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=customer_id" })
  customerId: string;
}

export class GetCustomersCustomerIdCreditsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCustomersCustomerIdCreditsPathParams;
}

export class GetCustomersCustomerIdCredits200ApplicationJSONData extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "balance" })
  balance: number;

  @SpeakeasyMetadata()
  @Expose({ name: "expiry_date" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  expiryDate: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id: string;

  @SpeakeasyMetadata()
  @Expose({ name: "per_unit_cost_basis" })
  perUnitCostBasis: string;
}

export class GetCustomersCustomerIdCredits200ApplicationJSONPaginationMetadata extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "has_more" })
  hasMore: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "next_cursor" })
  nextCursor?: string;
}

export class GetCustomersCustomerIdCredits200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: GetCustomersCustomerIdCredits200ApplicationJSONData })
  @Expose({ name: "data" })
  @Type(() => GetCustomersCustomerIdCredits200ApplicationJSONData)
  data: GetCustomersCustomerIdCredits200ApplicationJSONData[];

  @SpeakeasyMetadata()
  @Expose({ name: "pagination_metadata" })
  @Type(() => GetCustomersCustomerIdCredits200ApplicationJSONPaginationMetadata)
  paginationMetadata: GetCustomersCustomerIdCredits200ApplicationJSONPaginationMetadata;
}

export class GetCustomersCustomerIdCreditsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getCustomersCustomerIdCredits200ApplicationJSONObject?: GetCustomersCustomerIdCredits200ApplicationJSON;
}

File: go-client-sdk/pkg/models/operations/getcustomerscustomeridcredits.go
package operations

import (
	"time"
)

type GetCustomersCustomerIDCreditsPathParams struct {
	CustomerID string `pathParam:"style=simple,explode=false,name=customer_id"`
}

type GetCustomersCustomerIDCreditsRequest struct {
	PathParams GetCustomersCustomerIDCreditsPathParams
}

type GetCustomersCustomerIDCredits200ApplicationJSONData struct {
	Balance          float64   `json:"balance"`
	ExpiryDate       time.Time `json:"expiry_date"`
	ID               string    `json:"id"`
	PerUnitCostBasis string    `json:"per_unit_cost_basis"`
}

type GetCustomersCustomerIDCredits200ApplicationJSONPaginationMetadata struct {
	HasMore    bool    `json:"has_more"`
	NextCursor *string `json:"next_cursor,omitempty"`
}

type GetCustomersCustomerIDCredits200ApplicationJSON struct {
	Data               []GetCustomersCustomerIDCredits200ApplicationJSONData             `json:"data"`
	PaginationMetadata GetCustomersCustomerIDCredits200ApplicationJSONPaginationMetadata `json:"pagination_metadata"`
}

type GetCustomersCustomerIDCreditsResponse struct {
	ContentType                                           string
	StatusCode                                            int
	GetCustomersCustomerIDCredits200ApplicationJSONObject *GetCustomersCustomerIDCredits200ApplicationJSON
}
